### ISSUE TITLE
Unprotected/insecure create customer API

### ISSUE THREAD

**Issue by mischabraam** [magento/magento2#37374]
### Preconditions and environment

- Magento version 2.4.6

### Steps to reproduce

1. Create a new customer with unprotected POST REST API. As test body you can use the following.

```
POST http://www.magento2.test/rest/all/V1/customers
```
```
{
    "customer": {
        "group_id": 0,
        "email": "testing@spammers.com",
        "firstname": "Test",
        "lastname": "Spammer",
        "gender": 0,
        "store_id": 1,
        "website_id": 1
    }
}
```

### Expected result

The customer is not created, you should get an error like "auth is missing".

### Actual result

The customer is created blindly with some attributes having default values. This makes it possible for spammers to create customers endlessly.
See the following screenshots as proof of this insecure behavior.

![public-create-customer-1](https://user-images.githubusercontent.com/19824986/232027807-c2a98bc6-b25c-438a-8f6d-870f3c14214f.png)
![public-create-customer-2](https://user-images.githubusercontent.com/19824986/232027829-8512ba65-df68-4db3-a5e6-ce4a514de99b.png)
![public-create-customer-3](https://user-images.githubusercontent.com/19824986/232027841-d0417391-154b-4229-b3c9-86f25a554715.png)

What confuses me is that the GET and PUT API's seem to be secure. But why not the POST?

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Bravo** [comment]
@magento give me 2.4-develop instance

**Comment by mischabraam** [comment]
> Preventing anonymous access to these endpoints could cause third-party integrations to fail.

Very true. If you would secure this API by adding a resource in `webapi.xml` this will be not backwards compatible. However, that should not be a reason to not change it. :)

I think it's weird to force a bearer token (older versions) or oauth token for getting a customer, changing them, changing products, adding products, bot NOT for creating a customer. I can easily create thousands of spam customers accounts, reset their passwords and place fake orders with these fake accounts, fun fun fun for spammers. Don't you agree at least this is weird (despite what the docs state)?

**Comment by engcom-Bravo** [comment]
Hi @mischabraam,

Thanks for your update.

We are considering this as Feature request as it is an existing behaviour of Magento.

Thanks.

**Comment by maghamed** [comment]
@magento give me 2.4-develop instance

**Comment by DuckThom** [comment]
@mischabraam 

I think if you want to protect such routes from spammers you would need to look into using (re)captcha: https://developer.adobe.com/commerce/webapi/rest/use-rest/protected-endpoints/#recaptcha 

> I think it's weird to force a bearer token (older versions) or oauth token for getting a customer, changing them, changing products, adding products, bot NOT for creating a customer.

Correct me if I'm wrong but if an auth token was enforced on the create customer API, wouldn't that block the registration flow for things like standalone frontends that communicate with Magento using the REST API? 

As for the creation and updating of products, there's also a difference between admin/integration tokens and customer tokens. For example, customer tokens cannot do such actions whereas admin/integration tokens have access to basically anything you can do in the admin panel (assuming the permissions for the token are set correctly).

**Comment by DuckThom** [comment]
I agree that from an integration perspective it makes sense to require some sort of token to those endpoints, yes. 

However, I still don't think it would work for standalone frontends if the token is always required.

Let's say a token was required for the createCustomer call. When you submit a registration form directly to the Magento REST API, you'd also need send that token from the browser to Magento. This token could easily be read via the dev tools in the browser and could then be used to automate/spam requests once more (or worse, if it's an admin token, access the admin endpoints).

I think a solution could be to have a toggle or setting in the Magento admin panel which can be used to allow or disallow anonymous API's. 

- Allowed: you'd have the current situation which is useful for standalone frontends, or;
- Disallowed: only integrations can effectively access the API's. 

When disallowed, a standalone frontend could still be used, though you'd need something like NextJS where you could add the token via an API route: `browser > NextJS api route to add token > Magento API`. And even then, a bot could abuse the API route to still make the calls to Magento.

**Comment by mischabraam** [comment]
You would have to built in logic to accomplish that I guess, instead of a change in the `webapi.xml`. But if you think that's necessary, then yeah ok, sounds good. I would suggest then that the "disallowed" option would be the default? But I'm open for your opinion.

**Comment by DuckThom** [comment]
The default value kind of depends on if it should be a breaking change or not I suppose. 

If the default will be disallowed then it would mean that, without changing the configuration during an upgrade, the currently anonymous API routes will no longer be accessible.

Personally I think making it opt-in (so with `allowed` as default value) would be better/more upgrade friendly.

**Comment by mischabraam** [comment]
Correct, just like when the deprecated bearer token by default changed to not-accepted, unless you configure it manually to use it. That's my basis to suggest it here also.

**Comment by engcom-Bravo** [comment]
Hi @mischabraam,

Thanks for your reporting and collaboration.

We have tried to reproduce the issue in Latest 2.4-develop instance and the issue is no more reproducible.Kindly refer the screenshots.


<img width="874" alt="Image" src="https://github.com/user-attachments/assets/ee612a30-6116-48b1-9322-e59ccbb4e202" />



The Customer is not created.


Hence we are closing this issue.

Thanks.
